engineapi/engine.go

```diff
@@ -22,26 +24,30 @@ const (
 	ProcessStateInProgress = "in_progress"
 	ProcessStateComplete   = "complete"
 	ProcessStateError      = "error"
+
+	ErrNotImplement = "not implemented"
 )
 
 type EngineCollection struct{}
 
-type Engine struct {
+type EngineBinary struct {
 	name  string
 	image string
 	ip    string
 	port  int
 	cURL  string
 }
 
-func (c *EngineCollection) NewEngineClient(request *EngineClientRequest) (EngineClient, error) {
+func (c *EngineCollection) NewEngineClient(request *EngineClientRequest) (*EngineBinary, error) {
```

Comment:
Why not just return `EngineClientProxy` instead of `*EngineBinary`? so we can operate on interface instead of struct.

Reply:
This is used for fallback returns so we don't error out on missing methods.